cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -194,7 +194,9 @@ protected boolean isDefaultContextProperty(String property) {
       return true;
     }
     return ImmutableSet.of(DataprocConf.RUNTIME_JOB_MANAGER, BUCKET, DataprocConf.TOKEN_ENDPOINT_KEY,
-                           DataprocConf.ENCRYPTION_KEY_NAME).contains(property);
+                           DataprocConf.ENCRYPTION_KEY_NAME, DataprocConf.ROOT_URL,
+                           DataprocConf.COMPUTE_HTTP_REQUEST_CONNECTION_TIMEOUT,
```

Comment:
Instead of setting defaults in cdap-default.xml and specifying these keys here, it's simpler to have the defaulting logic in the DataprocConf class, similar to https://github.com/cdapio/cdap/blob/229c75adab836d38f03c008ae0779cf58f50fcb9/cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java#L580

cdap-default.xml should only contain CDAP configs anyway, not configs for extensions.

Unless there's a reason it needs to be here?

Reply:
Something like:

```
    String rootUrl = conf.getRootUrl() != null ? conf.getRootUrl() : ClusterControllerSettings.getDefaultEndpoint();
    String regionalEndpoint = conf.getRegion() + "-" + rootUrl;
```

Reply:
I agree. I wasn't aware of `ClusterControllerSettings.getDefaultEndpoint()` method in the Dataproc lib as previously also we were storing the root url in [CDAP](https://github.com/cdapio/cdap/pull/13795/files#diff-a396735e220d8347dce7a3a5c964a4fbe84034bb80da8fef95294e74e2ba9f84L110).
Will use the default endpoint as mentioned by @fernst and remove config from cdap-default.